src/main/java/io/confluent/connect/jdbc/JdbcSourceTask.java

```diff
@@ -257,6 +257,10 @@ public void stop() throws ConnectException {
   private void validateNonNullable(String incrementalMode, String table, String incrementingColumn,
                                    String timestampColumn) {
     try {
+      // Views typically will not expose NOT NULL constraints on their columns, so don't
+      // check for nullability.
+      if (JdbcUtils.isView(db, table)) return;
```

Comment:
Is that going to be an issue for compatibility as schemas evolve? (This partly depends on the converter you're using and if you want to enforce compatibility, of course.)